core/src/test-core/java/com/radixdlt/rev2/Manifest.java

```diff
@@ -0,0 +1,333 @@
+            CALL_METHOD Address("%s") "try_deposit_batch_or_abort" Expression("ENTIRE_WORKTOP");
+            """,
+            params.lockFeeLine(fromAccount),
+            params.encode(fromAccount),
+            params.encode(resource),
+            amount,
+            params.encode(toAccount));
+  }
+
+  public static Functions.Func1<Parameters, String> transferBetweenAccountsFeeFromReceiver(
+      ComponentAddress fromAccount,
+      ResourceAddress resource,
+      Decimal amount,
+      ComponentAddress toAccount) {
+    return (params) ->
+        String.format(
+            """
+            %s
+            CALL_METHOD Address("%s") "withdraw" Address("%s") Decimal("%s");
+            CALL_METHOD Address("%s") "deposit_batch" Expression("ENTIRE_WORKTOP");
+            """,
+            params.lockFeeLine(toAccount),
+            params.encode(fromAccount),
+            params.encode(resource),
+            amount,
+            params.encode(toAccount));
+  }
+
+  public static Functions.Func1<Parameters, String> transferBetweenAccountsFeeFromFaucet(
+      ComponentAddress fromAccount,
+      ResourceAddress resource,
+      Decimal amount,
+      ComponentAddress toAccount) {
+    return (params) ->
+        String.format(
+            """
+            %s
+            CALL_METHOD Address("%s") "withdraw" Address("%s") Decimal("%s");
+            CALL_METHOD Address("%s") "try_deposit_batch_or_abort" Expression("ENTIRE_WORKTOP");
+            """,
+            params.faucetLockFeeLine(),
+            params.encode(fromAccount),
+            params.encode(resource),
+            amount,
+            params.encode(toAccount));
+  }
+
+  public static Functions.Func1<Parameters, String> depositFromFaucet(ComponentAddress toAccount) {
+    return (params) ->
+        String.format(
+            """
+            %s
+            CALL_METHOD Address("%s") "free";
+            CALL_METHOD Address("%s") "try_deposit_batch_or_abort" Expression("ENTIRE_WORKTOP");
+            """,
+            params.faucetLockFeeLine(), params.encode(FAUCET), params.encode(toAccount));
+  }
+
+  public static Functions.Func1<Parameters, String> drainAccount(ComponentAddress account) {
+    // NOTE: A test relies on this only being able to be performed once per account
+    // So we transfer slightly less than the free XRD amount
+    final var accountToDrainTo =
+        Address.virtualAccountAddress(ECKeyPair.generateNew().getPublicKey());
+    return (params) ->
+        String.format(
+            """
+            %s
+            CALL_METHOD Address("%s") "withdraw" Address("%s") Decimal("%s");
+            CALL_METHOD Address("%s") "try_deposit_batch_or_abort" Expression("ENTIRE_WORKTOP");
+            """,
+            params.lockFeeLine(account),
+            params.encode(account),
+            params.encode(XRD),
+            ScryptoConstants.FREE_AMOUNT_FROM_FAUCET.subtract(Decimal.of(50)),
```

Comment:
```suggestion
            ScryptoConstants.FREE_AMOUNT_FROM_FAUCET.subtract(Decimal.of(1000)),
```

The last failing test (i.e. `later_rejected_transaction_should_not_linger_in_mempool()`) will pass if you adjust this line (I assume `50` was too low to account for lock fees etc.)

BTW the test improvements in this PR are unbelievable 👍